collections-csharp/HashtableInCSharp/Tests/HashtableInCSharpTest.cs

```diff
@@ -5,71 +5,120 @@ namespace Tests
+
+            Assert.IsType<Hashtable>(userHashtable);
+            Assert.Empty(userHashtable);
+        }
+
         [Fact]
-        public void GivenAnEmptyHashtable_WhenAddingToAHashtable_ThenReturnsFilledEmptyHashTable()
+        public void GivenHashtable_WhenCreatingHashtableWithInitialCapacity_ThenReturnsHashTable()
         {
-            Assert.IsType<Hashtable>(_users);
-            Assert.NotEmpty(_users);
+            var userHashtable = HashtableDemo.CreateHashTableWithInitialCapacity();
+
+            Assert.IsType<Hashtable>(userHashtable);
+            Assert.Empty(userHashtable);
+        }
+
+        [Fact]
+        public void GivenHashtable_WhenCreatingHashtableFromDictionary_ThenReturnsHashTable()
+        {
+            var userHashtable = HashtableDemo.CreateHashTableFromDictionary();
+
+            Assert.IsType<Hashtable>(userHashtable);
+            Assert.NotEmpty(userHashtable);
+            Assert.Equal(5, userHashtable.Count);
         }
 
         [Fact]
-        public void GivenAnEmptyHashtable_WhenRetrievingFromHashtable_ThenReturnsAUser()
+        public void GivenHashtable_WhenAddingSampleDataToHashtable_ThenReturnsHashTable()
         {
-            var user = _hashtableDemo.RetrieveAnElementFromHashTable(_users);
+            List<User> userList = new()
+            {
+                new User() { Id = 6, FirstName = "Judit", LastName = "Peter" },
+                new User() { Id = 7, FirstName = "Steve", LastName = "Billing" },
+                new User() { Id = 8, FirstName = "Goddy", LastName = "Opara" },
+            };
+
+            var populatedUserHashtable = HashtableDemo.AddSampleDataToHashTable(_userHashtable, userList);
+
+            Assert.IsType<Hashtable>(populatedUserHashtable);
+            Assert.NotEmpty(populatedUserHashtable);
+            Assert.Equal(8, populatedUserHashtable.Count);
+        }
+
+        [Fact]
+        public void GivenEmptyHashtable_WhenRetrievingElementFromHashtable_ThenReturnsUser()
+        {
+            var user = HashtableDemo.RetrieveElementFromHashTable(_userHashtable, 3);
             var userFirstName = user.FirstName;
             var userLastName = user.LastName;
 
-            Assert.Equal("Goddy", userFirstName);
-            Assert.Equal("Opara", userLastName);
+            Assert.Equal("Sam", userFirstName);
+            Assert.Equal("Manalt", userLastName);
         }
 
         [Fact]
-        public void GivenAnEmptyHashtable_WhenUpdatingElementInHashtable_ThenReturnsUpdatedHashTable()
+        public void GivenEmptyHashtable_WhenRetrievingAllElementFromHashtable_ThenReturnsAllUsers()
         {
-            var newUsers = _hashtableDemo.UpdateAnElementInHashTable(_users);
-            User user = (User)newUsers[1];
+            var users = HashtableDemo.RetrieveAllElementsFromHashTable(_userHashtable);
+
+            Assert.IsType<List<User>>(users);
+            Assert.NotEmpty(users);
+            Assert.Equal(5, users.Count);
```

Comment:
This assertion would be more maintainable and clearer if it said:
Assert.Equal(_userHashtable.Count, users.Count), as that is the actual assertion that is intended.